src/portable/extension.rs

```diff
@@ -53,27 +55,65 @@ fn get_local_instance(options: &Options) -> Result<InstanceInfo, anyhow::Error>
     Ok(inst)
 }
 
+#[tokio::main(flavor = "current_thread")]
+async fn _run_query<R>(connector: Connector, query: &str) -> Result<Vec<R>, anyhow::Error>
+where
+    R: QueryResult,
+{
+    let mut connection = connector.connect().await?;
+    let results = connection.query(query, &()).await?;
+    Ok(results)
+}
+
+type ExtensionInfo = (String, String);
+
 fn list(_: &ExtensionList, options: &Options) -> Result<(), anyhow::Error> {
-    let inst = get_local_instance(options)?;
-    let extension_loader = inst.extension_loader_path()?;
-    let output = run_extension_loader(&extension_loader, Some("--list-packages"), None::<&str>)?;
-    let value: serde_json::Value = serde_json::from_str(&output)?;
+    let extensions: Vec<ExtensionInfo> = (|| {
```

Comment:
Generally it would be better to create a new function rather than define a closure and immediately execute it.